programs/perpetuals/src/instructions/add_stake.rs

```diff
@@ -0,0 +1,272 @@
+//! AddStake instruction handler
+
+use {
+    crate::{
+        adapters,
+        adapters::SplGovernanceV3Adapter,
+        math, program,
+        state::{cortex::Cortex, perpetuals::Perpetuals, stake::Stake},
+    },
+    anchor_lang::prelude::*,
+    anchor_spl::token::{Mint, Token, TokenAccount},
+    solana_program::program_error::ProgramError,
+};
+
+#[derive(Accounts)]
+pub struct AddStake<'info> {
+    #[account(mut)]
+    pub owner: Signer<'info>,
+
+    #[account(
+        mut,
+        token::mint = lm_token_mint,
+        has_one = owner
+    )]
+    pub funding_account: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        token::mint = stake_reward_token_mint,
+        has_one = owner
+    )]
+    pub owner_reward_token_account: Box<Account<'info, TokenAccount>>,
+
+    // staked token vault
```

Comment:
I find it quite important, cause it can feel like the user account, idk I got confused a few times in my own code so I added it. Maybe I should rename instead?

Reply:
Yhea I'm puzzled over this one, i leave it as it is, if you wanna change let me know I change